### ISSUE TITLE
ERROR Error: NG05104

### ISSUE THREAD

**Issue by barthocologne** [angular/angular#50564]
### Which @angular/* package(s) are the source of the bug?

elements

### Is this a regression?

No

### Description

Using newest angular version (16.0.4) and exporting components as Custom Web Elements, the browser raises an error.

The Web Component works nevertheless.

**AppModule:**
```
import {DoBootstrap, NgModule} from '@angular/core';
import { BrowserModule } from '@angular/platform-browser';
import  { Injector} from '@angular/core';
import  { createCustomElement } from '@angular/elements';

import { AppComponent } from './app.component';
import {CustomerListComponent} from "./customer-list/customer-list.component";

@NgModule({
  declarations: [
    AppComponent,
    CustomerListComponent
  ],
  imports: [
    BrowserModule
  ],
  providers: [],
  bootstrap: [AppComponent, CustomerListComponent]
})
export class AppModule implements DoBootstrap{

  constructor(private injector: Injector) {

    const customers = createCustomElement(CustomerListComponent, { injector });
    customElements.define('customer-list-widget', customers);

  }

  ngDoBootstrap() {}

}
```

### Please provide the exception or error you saw

```true
ERROR Error: NG05104
    at Nc.selectRootElement (contact-list.min.js:3:111827)
    at EE (contact-list.min.js:3:63774)
    at jr.create (contact-list.min.js:3:63831)
    at e.bootstrap (contact-list.min.js:3:89100)
    at contact-list.min.js:3:87491
    at Array.forEach (<anonymous>)
    at e._moduleDoBootstrap (contact-list.min.js:3:87478)
    at contact-list.min.js:3:87180
    at v.invoke (contact-list.min.js:2:6564)
    at Object.onInvoke (contact-list.min.js:3:79818)
```

### Please provide the environment you discovered this bug in (run `ng version`)

```true
Angular CLI: 16.0.4
Node: 18.16.0
Package Manager: npm 9.5.1
OS: darwin x64

Angular: 16.0.4
... animations, cli, common, compiler, compiler-cli, core
... elements, forms, platform-browser, platform-browser-dynamic
... router

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1600.4
@angular-devkit/build-angular   16.0.4
@angular-devkit/core            16.0.4
@angular-devkit/schematics      16.0.4
@schematics/angular             16.0.4
rxjs                            7.8.1
typescript                      5.0.4
```

**Comment by JeanMeche** [comment]
Hi, do you use the `ngSkipHydration` in your `CustomerListComponent` ?

**Comment by barthocologne** [comment]
First try: I didn't even care about Hydration since I did not know about until I got that error.
Secont try: According  documentation I than added ngSkipHydration to the root template element
<div ngSkipHydration>...</div>

// https://angular.io/errors/NG0504

This error only comes up when I include the component as a web component. In an angular app everything works fine.

Hope this helps!?

**Comment by JeanMeche** [comment]
Oh sorry, this one is on me, I missed read the error code.

You issue lies with the bootstrap attribute. You shouldnt' put your webcomponent in it.

**Comment by barthocologne** [comment]
I removed WebComponents from bootstrap => Same error
I removed ALL Components (also AppComponent) from bootstrap => Error doesn't occur anymore.

Seems that a module with web components must not contain any bootstrap component?

As all examples I found related to WebComponent  require the component to be part of the bootstrap Part? Are there any changes related to this? Hydration is very new?...
Nevertheless, removing ALL bootstrap components solved my problem.
